api/v1alpha1/scaledjob_types.go

```diff
@@ -36,8 +36,14 @@ type ScaledJobSpec struct {
 	// +optional
 	EnvSourceContainerName string `json:"envSourceContainerName,omitempty"`
 	// +optional
-	MaxReplicaCount *int32          `json:"maxReplicaCount,omitempty"`
-	Triggers        []ScaleTriggers `json:"triggers"`
+	MaxReplicaCount *int32 `json:"maxReplicaCount,omitempty"`
+	// +optional
+	ScalingStrategy string `json:"scalingStrategy,omitempty"`
```

Comment:
I'd change this a little bit, so we don't have all settings in the top level:
```yaml
 pollingInterval: 5   # Optional. Default: 30 seconds
  cooldownPeriod: 30   # Optional. Default: 300 seconds
  maxReplicaCount: 100  # Optional. Default: 100
  successfulJobsHistoryLimit: 5
  failedJobsHistoryLimit: 5
  scalingStrategy: 
      strategy/type/mode?: "custom"                    # not sure about the name of this property, but probably strategy
      customScalingQueueLengthDeduction: 1
      customScalingRunningJobPercentage: "0.5"
```

Reply:
I love the idea. Done